### PR TITLE
Reformat, reorder, and add new tips to the teaching guide.

### LINKED ISSUE #112: Get instructors.md read for v5.3
There are some `FIXME` at `instructors.md`.

### PR DESCRIPTION
**Pull request by iglpdc**
This fixes #112.

### CODE REVIEW

**Comment by iglpdc** [comment]
**Don't merge this** before decide on #116. The file here should be adapted to the new numbering.

**Comment by iglpdc** [comment]
This depends on #116 and #124, and has to be merged afterward.

File: instructors.md [hunk 1]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
+    conflicts part of the lesson if the instructors want to inject some humor
+    into the room.
+
+*   If you don't have two projectors, have two instructors at the front of the
+    room.  Each instructor does their piece of the collaboration demonstration
+    on their own computer and then passes the projector cord back and forth
+    with the other instructor when it's time for them to do the other part of
+    the collaborative workflow.  It takes less than 10 seconds for each
+    switchover, so it doesn't interrupt the flow of the lesson.
+    And of course it helps to give each of the instructors a different-colored
```
Comment (r-gaia-cs) [inline instructors.md]: Colored hat is something that should be in every instructors bag. =) I need to find one for me.

File: instructors.md [hunk 2]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
+      work: `git clone https://github.com/vlad/planets.git planets-at-work`.
+    - At the office, you make a change, commit and push.  
+    - At your laptop repo, you (forget to pull and) make a change, commit and 
+      try to push. 
+    - `git pull` now and show the conflict.
+
+*   Learners usually forget to `git add` the file after fixing the conflict and
+    just (try to) commit. You can diagnose this with `git status`.
+
+*   Remember that you can discard one of the two parents of the merge:
```
Comment (r-gaia-cs) [inline instructors.md]: Maybe add that this is useful if you are working with binary files.

File: instructors.md [hunk 3]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
-
-1.  questions about who owns what, or can use what,
-    arise naturally once we start talking about
-    using public services like GitHub to store files;
-    and
-2.  the discussion gives learners a chance to catch their breath
-    after what is often a frustrating couple of hours.
+Version control might be the most important topic we teach, but Git is
+definitely the most complicated tool.  However, GitHub presently dominates the
+open software repository landscape, so we have to help novices learn just
```
Comment (msarahan) [inline instructors.md]: I would rephrase the last sentence here: 

However, GitHub presently dominates the open software repository landscape, so the time and effort required to teach fundamental Git is justified and worthwhile.

File: instructors.md [hunk 4]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
+enough Git to feel they can and should learn more on their own.
 
-## Teaching Notes
+There are many things we don't teach:
+
+- we don't teach branching, because despite being a powerful tool, it is an
```
Comment (msarahan) [inline instructors.md]: It is redundant to say "there are many things we don't teach:" and then start each bullet point with "we don't teach."  Can it be omitted on each bullet point?

- branching <not sure if the extra reasoning is necessary here>
- Git internals, such as hashes and commit objects

File: instructors.md [hunk 5]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
-*   Try to make it clear that Git and GitHub are not the same thing:
-    Git is an open source version control tool,
-    GitHub is a company that's built a really useful website on top of it.
+Instead to try convince them that version control is useful for researchers 
+working in teams or not, because it is:
 
-*   Ask, "Who uses 'undo' in their editor?"
-    They say, "'Undo' is the simplest form of version control."
+- a better way to "undo" changes,
+- a better way to collaborate than mailing files back and forth, and
```
Comment (msarahan) [inline instructors.md]: Commas unnecessary in bulleted list

File: instructors.md [hunk 6]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
+    and, "come on this really has to be the last version" to motivate version
+    control as a better way to collaborate and as a better way to back work up.
 
 ## [Setting Up Git](01-setup.html)
 
-We suggest instructors and students use `nano` as the text editor for this lessons because:
+*   We suggest instructors and students use `nano` as the text editor for this
+    lessons because:
+    
+    - it runs in all three major operating systems,
```
Comment (msarahan) [inline instructors.md]: no commas at end of bulleted list items

File: instructors.md [hunk 7]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
+*   The most common mistake is that learners `push` before `pull`ing. If they
+    `pull` afterward, they may get a conflict.
+
+*   Conflicts, sometimes weird, will start to arise. Stay tight: conflicts are
+    next.
 
 ## [Conflicts](07-conflict.html)
 
-FIXME
+*   Expect the learners to make mistakes. Expect *you* to make mistakes. This
```
Comment (msarahan) [inline instructors.md]: *you* => *yourself*

File: instructors.md [hunk 8]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
 
-FIXME
 
 ## [Licensing](09-licensing.html)
 
-FIXME
+We teach about licensing because:
+
+- questions about who owns what, or can use what, arise naturally once we
+  start talking about using public services like GitHub to store files; and
```
Comment (msarahan) [inline instructors.md]: no "; and" at end of bulleted list line.  This might be better if simply converted to a paragraph:


We discuss licensing because questions about who owns what, or can use what, arise naturally once we
start talking about using public services like GitHub to store files.  This discussion gives learners a chance to catch their breath after what is often a frustrating couple of hours.

File: instructors.md [hunk 9]
```diff
@@ -7,196 +7,248 @@ subtitle: Instructor's Guide
...
+  start talking about using public services like GitHub to store files; and 
+- the discussion gives learners a chance to catch their breath after what
+  is often a frustrating couple of hours.
 
 ## [Hosting](10-hosting.html)
 
-FIXME
+A common concern for learners is having their work publicly available on
+Github.  While we encourage open science, sometimes private repos are the
+only choice. It's always interesting to mention the options to have
```
Comment (msarahan) [inline instructors.md]: options of having them, and benefits of having them: you can still tie online CI services to private repositories.  Taking advantage of these existing build connections can save a lot of time and effort.

**Comment by msarahan** [comment]
I have made lots of comments that don't necessarily pertain directly to this commit.  If they're better as a separate PR, I will do so once this PR is sorted.

File: instructors.md (hunk 4 above)
Comment (iglpdc) [inline instructors.md]: Thanks, I'm gonna change the whole paragraph, it doesn't make sense. The next refers to "them" (the novice learners), which is not longer here. I'm thinking in:

"Because of this complexity, we don't teach novice learners about many
interesting topics, such as branching, hashes, or commit objects."

File: instructors.md (hunk 8 above)
Comment (iglpdc) [inline instructors.md]: Yep.

File: instructors.md (hunk 9 above)
Comment (iglpdc) [inline instructors.md]: @msarahan do you wanna make a new PR with this last comment? That'd be great! I don't know much about CI. I'll leave it to you whether this is a thing:

- to leave here (mostly a hint for the instructors that learners don't necessarily see)
- to put in the 10-hosting.md (so it's part of the lesson.)
- to put in discussion.md (it's something at a novice level, but takes too much time to explain)

**Comment by iglpdc** [comment]
Thanks, @msarahan ! Great review.

File: instructors.md (hunk 3 above)
Comment (iglpdc) [inline instructors.md]: Changed.

File: instructors.md (hunk 2 above)
Comment (iglpdc) [inline instructors.md]: Ok, thanks!

**Comment by iglpdc** [comment]
Hey, @msarahan, I'm not sure about the commas and the bullet points. In the two cases that we have left, I think punctuation inside the list is OK. The rule that I've found is "introduce with a colon and punctuate as usual" or "avoid the colon, capitalize, and don't punctuate the bullet points". The second is only when the introductory sentence is complete on their own [1].

So, I'm +1 for leaving the commas and removing the colons. Do you have a better reference? 

@amyrbrown do you wanna weight in?

It's not a big deal in any case: mostly curiosity now. 

[1] http://www.princeton.edu/communications/services/editorial/resources/style-guide/#comp00004e94bd3d0000005d595a6a

**Comment by msarahan** [comment]
Yes, that Princeton style guide does things the way that they are here - but the TLDR I gather from the end is: it doesn't matter, just be consistent.  This is not the way I would do it, but I won't argue with style guides!  I had enough of that in high school English class.

**Comment by iglpdc** [comment]
Ok. Let's follow the guide then. We can always change it later if someone else offers a best solution. I don't want the PR to be waiting for a small thing.

Thanks again for the review, @msarahan and @r-gaia-cs

File: instructors.md [hunk 10]
```diff
@@ -7,196 +7,245 @@ subtitle: Instructor's Guide
...
+fundamental Git is justified and worthwhile.
 
-## Teaching Notes
+Because of this complexity, we don't teach novice learners about many
+interesting topics, such as branching, hashes, and commit objects.
 
-*   Try to make it clear that Git and GitHub are not the same thing:
-    Git is an open source version control tool,
-    GitHub is a company that's built a really useful website on top of it.
+Instead to try convince them that version control is useful for researchers
```
Comment (swaldman3) [inline instructors.md]: s/Instead to try/Instead we try to/

File: instructors.md [hunk 11]
```diff
@@ -7,196 +7,245 @@ subtitle: Instructor's Guide
...
+    master` a bit later will not give easily comprehensible results.  It also
+    makes it look like commits can be lost.  To "fix" a "detached HEAD", simply
+    `git checkout master`.
+
+*   This is a good moment to show a log within a Git GUI. If you skip it
+    because you're short on time, show it once in Github.
 
 ## [Ignoring Things](06-ignore.html)
+
+Just remember that you can use regular expressions in `.gitignore`.
```
Comment (swaldman3) [inline instructors.md]: How many of our students know regexps?

File: instructors.md [hunk 12]
```diff
@@ -7,196 +7,245 @@ subtitle: Instructor's Guide
...
+    with the other instructor when it's time for them to do the other part of
+    the collaborative workflow.  It takes less than 10 seconds for each
+    switchover, so it doesn't interrupt the flow of the lesson.
+    And of course it helps to give each of the instructors a different-colored
+    hat, or put different-colored sticky notes on their foreheads.
+
+*   If you're the only instructor, the best way to create is clone the two
+    repos in your Desktop, but under different names, e.g., pretend one is your
+    computer at work: 
+
```
Comment (swaldman3) [inline instructors.md]: Suggest maybe using two computers, even if only one instructor? Especially if two projectors are available, this might make it considerably more obvious what is being done from each version of the repo.

File: instructors.md [hunk 13]
```diff
@@ -7,196 +7,245 @@ subtitle: Instructor's Guide
...
 
-FIXME
+*   Expect the learners to make mistakes. Expect *yourself* to make mistakes. This
+    happens because it is late in the lesson and everyone is tired.
 
-## [Open Science](10-open.html)
+*   If you're the only instructor, the best way to create a conflict is:
 
-FIXME
+    - Clone your repo in a different directory, pretending is your computer at
```
Comment (swaldman3) [inline instructors.md]: This duplicates some of the discussion in "collaboration" above. Suggest that the instructor here simply adopts whatever approach was used in the last section.

Comment (iglpdc) [inline instructors.md]: Yes, I agree. But I put this line, because it's actually hard to create a conflict if you are the only instructor. I think it's just is late in the lesson, so I wanted to have a reminder on how to do it.

File: instructors.md (hunk 11 above)
Comment (iglpdc) [inline instructors.md]: None, probably, but this is for the instructors. I'll change it to "wildcards and regular expressions to ignore a particular set of files". It's a common question whether or how you can ignore a particular set of files, say all executables in a `lib/` folder.

File: instructors.md (hunk 12 above)
Comment (iglpdc) [inline instructors.md]: Yes, but it's unlikely to have a spare laptop properly configured. Most of the time is just some problem with the room: the podium doesn't fit two people or two laptops, the projector is too slow to switch between laptops or something like this.

File: instructors.md (hunk 10 above)
Comment (iglpdc) [inline instructors.md]: Thanks!

**Comment by iglpdc** [comment]
Thanks @swaldman3 ! I've added your corrections in a separate PR #136.
